rotkehlchen/tests/unit/decoders/test_rainbow.py

```diff
@@ -0,0 +1,225 @@
+def test_rainbow_swap_token_to_eth(ethereum_inquirer, ethereum_accounts):
+    tx_hash = deserialize_evm_tx_hash('0xd34d7393151a3c0f2d23d0df4d8f0b4a000be7613277e831ef0860191e68f855')  # noqa: E501
+    events, _ = get_decoded_events_of_transaction(evm_inquirer=ethereum_inquirer, tx_hash=tx_hash)
+    user_address = ethereum_accounts[0]
+    timestamp = TimestampMS(1741892171000)
+    swap_amount, received_amount, gas_fees, fee_amount = '1010.887928111872496631', '0.089068967427375408', '0.000147805218572058', '0.00089852786738257'  # noqa: E501
+    expected_events = [EvmEvent(
+        tx_hash=tx_hash,
+        sequence_index=0,
+        timestamp=timestamp,
+        location=Location.ETHEREUM,
+        event_type=HistoryEventType.SPEND,
+        event_subtype=HistoryEventSubType.FEE,
+        asset=A_ETH,
+        amount=FVal(gas_fees),
+        location_label=user_address,
+        notes=f'Burn {gas_fees} ETH for gas',
+        counterparty=CPT_GAS,
+    ), EvmEvent(
+        tx_hash=tx_hash,
+        sequence_index=1,
+        timestamp=timestamp,
+        location=Location.ETHEREUM,
+        event_type=HistoryEventType.TRADE,
+        event_subtype=HistoryEventSubType.SPEND,
+        asset=A_BUOY,
+        amount=FVal(swap_amount),
+        location_label=user_address,
+        notes=f'Swap {swap_amount} BOOE in Rainbow Swap',
+        address=RAINBOW_ROUTER_CONTRACT,
+        counterparty=CPT_RAINBOW_SWAPS,
+    ), EvmEvent(
+        tx_hash=tx_hash,
+        sequence_index=2,
+        timestamp=timestamp,
+        location=Location.ETHEREUM,
+        event_type=HistoryEventType.TRADE,
+        event_subtype=HistoryEventSubType.RECEIVE,
+        asset=A_ETH,
+        amount=FVal(received_amount),
+        location_label=user_address,
+        notes=f'Receive {received_amount} ETH as the result of a Rainbow Swap',
+        address=RAINBOW_ROUTER_CONTRACT,
+        counterparty=CPT_RAINBOW_SWAPS,
+    ), EvmEvent(
+        tx_hash=tx_hash,
+        sequence_index=333,
+        timestamp=timestamp,
+        location=Location.ETHEREUM,
+        event_type=HistoryEventType.SPEND,
+        event_subtype=HistoryEventSubType.FEE,
+        asset=A_ETH,
+        amount=FVal(fee_amount),
+        location_label=user_address,
+        notes=f'Spend {fee_amount} ETH as Rainbow Swap fees',
+        counterparty=CPT_RAINBOW_SWAPS,
+        address=RAINBOW_ROUTER_CONTRACT,
+    )]
+
+    assert expected_events == events
+
+
+@pytest.mark.vcr(filter_query_parameters=['apikey'])
+@pytest.mark.parametrize('ethereum_accounts', [['0xB80177697e160eFC91d4B1ec295ABE6Ce0c0Fe1f']])
+def test_rainbow_swap_token_to_token(ethereum_inquirer, ethereum_accounts):
+    tx_hash = deserialize_evm_tx_hash('0x4f85ee11bbc401240755b083106e0811e6b60ef7972063a51596934cbb6ed43f')  # noqa: E501
+    events, _ = get_decoded_events_of_transaction(evm_inquirer=ethereum_inquirer, tx_hash=tx_hash)
```

Comment:
this transaction is not decoded properly. You have an approval event in between the trade events. It has to be `spend -> receive -> fee` without other events between them